docs/api/topics/app_validation.rst

```diff
@@ -1,112 +0,0 @@
-.. _app_validation:
```

Comment:
I swear I'm done moving this file around. I initially wanted to share it between extensions and apps but decided against it to I'm moving it back to the `addons_submission` topic.